flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala

```diff
@@ -24,4 +24,19 @@ package org.apache.flink.table.functions
   * User-defined functions must have a default constructor and must be instantiable during runtime.
   */
 trait UserDefinedFunction {
```

Comment:
yes, It works! please refer to: http://alvinalexander.com/scala/how-to-wrap-scala-traits-used-accessed-java-classes-methods